hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactFunction.java

```diff
@@ -130,4 +143,38 @@ private HoodieWriteConfig reloadWriteConfig() throws Exception {
   public void setExecutor(NonThrownExecutor executor) {
     this.executor = executor;
   }
+
+  private void sendEventByMailBox(Collector<CompactionCommitEvent> collector, CompactionCommitEvent commitEvent) {
+    mailboxExecutorAdapter.execute(() -> collector.collect(commitEvent),
+            "Send compaction commit event by mailbox.");
+  }
```

Comment:
Thanks, we may need to figure out more simpler way for `collector` concurrency.

Reply:
Just pushed a new implementaion using AsynFunction of flink.
And fixed the same potential problem in clustering operator.